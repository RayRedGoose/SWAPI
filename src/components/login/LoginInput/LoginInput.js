import React from 'react';
import PropTypes from 'prop-types'
import './LoginInput.scss'

const LoginInput = ({type, value, toggleChangeState}) => {
  const handleChange = (event) => {
    toggleChangeState(type, event.target.value)
  }

  return (
    <section className="login-input">
      <label htmlFor={type.replace(' ', '-')}>{`your ${type}`}</label>
      <input
        id={type.replace(' ', '-')}
        value={value}
        placeholder={`enter your ${type}`}
        onChange={handleChange}/>
    </section>
  )
}

export default LoginInput

LoginInput.propTypes = {
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  toggleChangeState: PropTypes.func.isRequired
}
