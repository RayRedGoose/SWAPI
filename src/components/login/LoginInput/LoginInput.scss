$border-color: #F1AA61;

.login-input {
  display: flex;
  flex-direction: column;
  margin: 20px auto;
  width: 70%;
  label {
    font-size: 18px;
    text-align: left;
    text-transform: uppercase;
    margin-bottom: 10px;
  }
  input {
    font-size: 16px;
    padding: 0.6em;
    padding-left: 20px;
    border: 3px solid $border-color;
  }
}
