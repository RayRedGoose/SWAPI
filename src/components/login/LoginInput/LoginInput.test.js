import React from 'react'
import {shallow} from 'enzyme'
import LoginInput from './LoginInput'

describe("LoginInput", () => {
  let mockProps, input
  const toggleChangeState = jest.fn()

  beforeEach(() => {
    mockProps = {
      key: 'name',
      type: 'name',
      value: 'Skywalker',
      toggleChangeState: toggleChangeState
    }

    input = shallow(<LoginInput {...mockProps} />)
  })

  it ('should match the snapshot with all data passed in correctly', () => {
    expect(input).toMatchSnapshot()
  })

  it('should call toggleChangeState prop when input is changed', () => {
    input.find('input').simulate(
      'change',
      { target: { value: 'Yoda'}
    })

    expect(toggleChangeState).toHaveBeenCalled()
  })
})
