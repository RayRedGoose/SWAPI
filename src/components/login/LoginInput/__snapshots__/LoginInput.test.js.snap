// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginInput should match the snapshot with all data passed in correctly 1`] = `
<section
  className="login-input"
>
  <label
    htmlFor="name"
  >
    your name
  </label>
  <input
    id="name"
    onChange={[Function]}
    placeholder="enter your name"
    value="Skywalker"
  />
</section>
`;
